plugins/modules/aci_l3out_extepg_to_contract.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, fn ln (@kudtarkar1) <@.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_l3out_extepg_to_contract:
+short_descrption: Bind External End Point Groups to Contracts 
+descrption: 
+- Bind ExtEPGs to Contracts on ACI fabrics.
+Version_added: 
+Options:
+  tenant:
+    descrption:
+    - Name of existing tenant.
+    type: str
+    aliases: [tenant_name]
+  l3Out:
+    descrption:
+    - The name of the L3Outs
+    type: str
+    aliases: [l3out_name]
+  extepg:
+    descrption:
+    - The name of the external end point groups
+    type: str
+    aliases: [extepg_name]
+  contract:
+    descrption:
+    - The name of the contract
+    type: str
+  contract_type:
+    descrption:
+    - The type of contract(provider or consumer)
+  state:
+    descrption:
+    - Use pesent or absent for adding and removing
+    - Use query for listing an object or multiple ojbects
+    type: str
+    choices: [present, absent, query]
+extends_documentation_fragment: aci
+note:
+- The tenant, l3Out, extepg, contract should exist before using this module
+'''
+
+EXAMPLES = r'''
+- name: Bind External End Point Groups to Contracts 
+  cisco.aci.aci_l3out_epg_to_contract:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: Auto-Demo
+    l3Out: l3out
+    extepg : testEpg
+    contract: contract1
+    contract_type: provider
+    state: present
+  delegate_to: localhost
+
+- name: Remove existing contract to External End Point Groups
+  cisco.aco.aci_l3out_epg_to_contract:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: Auto-Demo
+    l3Out: l3out
+    extepg : testEpg
+    contract: contract1
+    contract_type: provider
+    state: absent
+  delegate_to: localhost
+        
+- name: Query the OOB mgmt ipv4 address  
+  cisco.aci.aci_l3out_epg_to_contract:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: Auto-Demo
+    l3Out: l3out
+    extepg : testEpg
+    contract: contract1
+    contract_type: provider
+    state: query
+  delegate_to: localhost
+  register: query_result
```

Comment:
I suppose that this can also be used to query all contracts, can you add an example of it?